### ISSUE TITLE
pyknxcommunicator uses about 100%CPU

### ISSUE THREAD

**Issue by MelbarKasom** [2franix/pyknx#5]
Hi 2franix,

thank you for your helpfull package. After a few days of usage i noticed that my Raspi's CPU got very warm. The reason was "pyknxcommunicat" which was using about 100% CPU.

Last start of the deamon was:
21.08.2015 00:01:59 pyknxcommunicator.py -d /var/lib/linknx/pyknx.conf
==> pyknx.conf is emtpy, because i currently only use "pyknxwrite.py" for sending temperatures to the bus.

22.08.2015 09:32:12 - pi@raspi:~$ top
top - 09:32:12 up 18 days, 10:44,  1 user,  load average: 1,07, 1,06, 1,06
Tasks: 115 total,   1 running, 114 sleeping,   0 stopped,   0 zombie
%Cpu(s): 20,1 us,  5,4 sy,  0,0 ni, 74,5 id,  0,0 wa,  0,0 hi,  0,0 si,  0,0 st
KiB Mem:    949408 total,   930040 used,    19368 free,   123008 buffers
KiB Swap:  3873232 total,        0 used,  3873232 free,   549964 cached

  PID USER      PR  NI  VIRT  RES  SHR S  %CPU %MEM    TIME+  COMMAND
  907 root      20   0 25300  13m 4080 S  99,9  1,5   2007:09 pyknxcommunicat

Is this a known bug ? Any idea what goes wrong ?

Has pyknxcommunicator problems when linknx/eibd is restarted in background without restart of pyknxcommunicator ?

Best regards,
Thorsten

**Comment by 2franix** [comment]
Hi Thorsten,

I cannot explain this excessive CPU consumption for the moment. But let me point out a few remarks:
- if you only use pyknxread.py or pyknxwrite.py, there is no need to start a pyknxcommunicator instance at all. The communicator is required to get events from linknx whenever an object value changes. That's all. Both pyknxread.py and pyknxwrite.py are simple client scripts for linknx.
- to better understand what is going on next time you can reproduce this issue, and assuming you will still run pyknxcommunicator after reading my first bullet point, please add --log-file and -v debug options in order to save a verbose log to a file. Attach the file to your message if you reproduce the problem.
- pyknxcommunicator has never been designed to work with an empty user script, though that should not be a big deal.
- restarting the communicator after restarting linknx is definitely a good practice. Keeping it alive forever should not lead to real issues, though. The first idea I can come up with is that some client calls from the user script may fail while linknx is being restarted. But since you do not seem to have any client code hosted by the communicator, that looks like a bad lead to follow.

The bottom line is: reconsider the real need to run a communicator instance in your case :)

Hope it helps...
Cyrille

**Comment by 2franix** [comment]
Hi Thorsten,

Any update? Does this solve your problem?
If so, please close this issue. Otherwise, let me know.

Thanks,
Cyrille

**Comment by MelbarKasom** [comment]
Hi Cyrille,

sorry, i've been on holiday. 

I had the problem two times in the first week of use. Since then I was not able to reproduce the issue, because it didn't happen again ..... 

You can close this issue, and I will stop the communicator instance until I need it.

Thanks for your support. I'll give you a short notice when it happens again.

best regards,

Thorsten
